From 2ef6a51a548acbb136ea3981a65e3115911ee4b1 Mon Sep 17 00:00:00 2001
From: Greg Thain <gthain@cs.wisc.edu>
Date: Tue, 26 Apr 2016 17:02:27 -0500
Subject: [PATCH 2/2] Set docker group id to the same that vanilla universe
 would use #5649

(cherry picked from commit 6db262581aa7d53776fea7d3e7cf9f358420db04)
---
 src/condor_starter.V6.1/docker-api.cpp | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/condor_starter.V6.1/docker-api.cpp b/src/condor_starter.V6.1/docker-api.cpp
index b4455e3..7885a57 100644
--- a/src/condor_starter.V6.1/docker-api.cpp
+++ b/src/condor_starter.V6.1/docker-api.cpp
@@ -114,19 +114,24 @@ int DockerAPI::run(
 	// Run with the uid that condor selects for the user
 	// either a slot user or submitting user or nobody
 	uid_t uid = 0;
+	uid_t gid = 0;
 
 	// Docker doesn't actually run on Windows, but we compile
 	// on Windows because...
 #ifndef WIN32
 	uid = get_user_uid();
+	gid = get_user_gid();
 #endif
 	
-	if (uid == 0) {
+	if ((uid == 0) || (gid == 0)) {
 		dprintf(D_ALWAYS|D_FAILURE, "Failed to get userid to run docker job\n");
 		return -9;
 	}
+
 	runArgs.AppendArg("--user");
-	runArgs.AppendArg(uid);
+	std::string uidgidarg;
+	formatstr(uidgidarg, "%d:%d", uid, gid);
+	runArgs.AppendArg(uidgidarg);
 
 	// Run the command with its arguments in the image.
 	runArgs.AppendArg( imageID );
-- 
2.5.5

